beetsplug/play.py

```diff
@@ -155,3 +190,21 @@ def _create_tmp_playlist(self, paths_list):
             m3u.write(item + b'\n')
         m3u.close()
         return m3u.name
+
+    def before_choose_candidate_listener(self, session, task):
```

Comment:
Argh; I wish we had chosen a better name than `before_choose_candidate`. :scream: But that's hardly your fault.